src/block/Door.php

```diff
@@ -176,4 +168,34 @@ public function getAffectedBlocks() : array{
 	private function canBeSupportedAt(Block $block) : bool{
 		return $block->getAdjacentSupportType(Facing::DOWN)->hasEdgeSupport();
 	}
+
+	public function onProjectileInteraction(Projectile $projectile) : void{
+		if($projectile instanceof WindCharge) {
+			if($this->getTypeId() === BlockTypeIds::IRON_DOOR) {
+				return;
+			}
+
+			// This is to avoid calling Door::onProjectileInteraction() twice due to two tiles.
+			if($this->top) {
+				return;
+			}
+
+			$this->toggle();
+		}
+	}
+
+	public function toggle() : void {
```

Comment:
This should be `private` or `protected`, making it public has implications beyond this PR's scope